cannon/cmd/witness.go

```diff
@@ -0,0 +1,53 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/urfave/cli/v2"
+)
+
+var (
+	WitnessInputFlag = &cli.PathFlag{
+		Name:      "input",
+		Usage:     "path of input JSON state.",
+		TakesFile: true,
+		Required:  true,
+	}
+	WitnessOutputFlag = &cli.PathFlag{
+		Name:      "output",
+		Usage:     "path to write binary witness.",
```

Comment:
My thinking is that we want the exact binary output since we're going to hash it and for the go side this is definitely easiest since we can just read the file and use the result directly.  I assume we can just read a binary file as part of the contract deployments scripts but its possible it will come unstuck there and we'll have to add a json wrapper.